packages/data/src/lib/data-view/data-view.ts

```diff
@@ -0,0 +1,132 @@
+import { CartoError, WithEvents } from '@carto/toolkit-core';
+import { Layer, CARTOSource } from '@carto/toolkit-viz';
+import { groupValuesByAnotherColumn } from '../aggregations/grouping';
+import {
+  applyAggregations,
+  AggregationTypes
+} from '../aggregations/aggregations';
+
+export class DataView extends WithEvents {
+  private source: CARTOSource | Layer;
+  private column: string;
+
+  constructor(source: CARTOSource | Layer, column: string) {
+    super();
+    validateParameters(source, column);
+
+    this.source = source;
+    this.column = column;
+
+    this.bindEvents();
+  }
+
+  private bindEvents() {
+    this.registerAvailableEvents(['dataUpdate']);
+
+    if (this.source instanceof Layer) {
+      this.source.on('viewportLoad', () => {
+        this.onDataUpdate();
+      });
+    }
+  }
+
+  private onDataUpdate() {
+    this.emitter.emit('dataUpdate');
+  }
+
+  async aggregate(...operations: AggregationTypes[]) {
```

Comment:
What's the main advantage of being able to provide different operations on every call...? I mean. should not we simplify it and just pass it as options to the constructor? That's the carto.js model, and of course we can change it but... Should the same dataview change its aggregation ops during its life cycle? In what cases? I can't think of any example...